lucene/core/src/java/org/apache/lucene/index/FieldInfos.java

```diff
@@ -481,205 +433,102 @@ synchronized int addOrGet(
+      verifySameVectorOptions(
+          fieldName, props.numDimensions, props.searchStrategy, vectorDimension, searchStrategy);
     }
 
-    synchronized void verifyConsistentVectorProperties(
-        Integer number,
-        String name,
-        int numDimensions,
-        VectorValues.SearchStrategy searchStrategy) {
-      if (name.equals(numberToName.get(number)) == false) {
-        throw new IllegalArgumentException(
-            "field number "
-                + number
-                + " is already mapped to field name \""
-                + numberToName.get(number)
-                + "\", not \""
-                + name
-                + "\"");
-      }
-      if (number.equals(nameToNumber.get(name)) == false) {
-        throw new IllegalArgumentException(
-            "field name \""
-                + name
-                + "\" is already mapped to field number \""
-                + nameToNumber.get(name)
-                + "\", not \""
-                + number
-                + "\"");
-      }
-      FieldVectorProperties props = vectorProps.get(name);
-      if (props != null) {
-        if (props.numDimensions != numDimensions) {
-          throw new IllegalArgumentException(
-              "cannot change vector dimension from "
-                  + props.numDimensions
-                  + " to "
-                  + numDimensions
-                  + " for field=\""
-                  + name
-                  + "\"");
-        }
-        if (props.searchStrategy != searchStrategy) {
-          throw new IllegalArgumentException(
-              "cannot change vector search strategy from "
-                  + props.searchStrategy
-                  + " to "
-                  + searchStrategy
-                  + " for field=\""
-                  + name
-                  + "\"");
-        }
-      }
+    /**
+     * Returns true if the {@code fieldName} exists in the map and is of the same {@code dvType},
+     * and it is docValues only field.
+     */
+    synchronized boolean containsDvOnlyField(String fieldName, DocValuesType dvType) {
+      if (nameToNumber.containsKey(fieldName) == false) return false;
+      if (dvType != docValuesType.get(fieldName)) return false;
+      FieldDimensions fdimensions = dimensions.get(fieldName);
+      if (fdimensions != null && fdimensions.dimensionCount != 0) return false;
+      IndexOptions ioptions = indexOptions.get(fieldName);
+      if (ioptions != null && ioptions != IndexOptions.NONE) return false;
+      FieldVectorProperties fvp = vectorProps.get(fieldName);
+      if (fvp != null && fvp.numDimensions != 0) return false;
+      return true;
     }
 
     /**
-     * Returns true if the {@code fieldName} exists in the map and is of the same {@code dvType}.
+     * Construct a new FieldInfo based on the options in global field numbers This method is not
+     * synchronized as all the options it uses are not modifiable.
```

Comment:
From the point of view `FieldInfos.FieldNumbers`,  doc values type is not modifiable and set only once on adding this field to global field numbers.

This method `constructFieldInfo` also checks the the provided in arguments `dvType` matches the one stored in the global field numbers for this field.  So I am not sure how it can be modifiable.

Please let me know if I am missing something.